core/chainio/retryable.go

```diff
@@ -203,10 +258,19 @@ func SubscribeToNewTasksV2Retryable(
 	newTaskCreatedChan chan *servicemanager.ContractAlignedLayerServiceManagerNewBatchV2,
 	batchMerkleRoot [][32]byte,
 ) (event.Subscription, error) {
+	return retry.RetryWithData(SubscribeToNewTasksV2(opts, serviceManager, newTaskCreatedChan, batchMerkleRoot), retry.DefaultRetryConfig())
+}
+
+func SubscribeToNewTasksV3(
+	opts *bind.WatchOpts,
+	serviceManager *servicemanager.ContractAlignedLayerServiceManager,
+	newTaskCreatedChan chan *servicemanager.ContractAlignedLayerServiceManagerNewBatchV3,
+	batchMerkleRoot [][32]byte,
+) func() (event.Subscription, error) {
 	subscribe_func := func() (event.Subscription, error) {
-		return serviceManager.WatchNewBatchV2(opts, newTaskCreatedChan, batchMerkleRoot)
+		return serviceManager.WatchNewBatchV3(opts, newTaskCreatedChan, batchMerkleRoot)
```

Comment:
We should add the fallback here

Reply:
Fallback logic is handled in the larger calling context of SubscribeToNewTaskV3.